### PR TITLE
Adding chocV keyboard

### PR DESCRIPTION
**Pull request by brickbots**
## Description

Add configuration and default keymap for the ChocV keyboard.  PCB's and details available at:
https://github.com/brickbots/chocV

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [X] Keyboard (addition or update)
- [X] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Checklist

- [X] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [X] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [X] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [X] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by tzarc** [comment]
I'd suggest going through the PR checklist that you've already ticked as "complete" -- you've missed stuff.

**Comment by brickbots** [comment]
> I'd suggest going through the PR checklist that you've already ticked as "complete" -- you've missed stuff.

Thank you, it's been a while since my last PR and I did forget a couple things and I see some new items.  Sorry for the sloppiness here, I'll tidy this up and push changes soon.

**Comment by brickbots** [comment]
Thanks again @tzarc for the gentle correction here :-)  I believe I've fixed up the checklist issues!

File: keyboards/chocv/readme.md [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+pro-micro footprint MCU
+* Hardware Availability: [The chocV github repo](https://github.com/brickbots/chocV)
+
+Make example for this keyboard (after setting up your build environment):
+
+    make chocv:manna-habour_miryoku
+
+Flashing example for this keyboard:
+
+    make chocv:manna-habour_miryoku:flash
```
Comment (tzarc) [inline keyboards/chocv/readme.md:19]:
```suggestion
    make chocv:default

Flashing example for this keyboard:

    make chocv:default:flash
```

File: keyboards/chocv/keymaps/default/readme.md [hunk 2]
```diff
@@ -0,0 +1,15 @@
+# The default keymap for ChocV is Manna Harbour Miryoku
+
+For details and building instructions and options please see the
+[The official Miryoku QMK
+Repo](https://github.com/manna-harbour/miryoku_qmk/tree/miryoku/users/manna-harbour_miryoku)
+
+But the tl;dr version is
+
+make chocv:manna-habour_miryoku
```
Comment (tzarc) [inline keyboards/chocv/keymaps/default/readme.md:9]:
```suggestion
make chocv:default
```

File: keyboards/chocv/keymaps/default/config.h [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+// place overrides here
```
Comment (tzarc) [inline keyboards/chocv/keymaps/default/config.h:19]: Let's remove this file, it serves no purpose.

**Review by tzarc — COMMENTED** [review]
The keymap you were specifying doesn't exist in the PR, and should be pointing to `:default` anyway.

Other than that, looks good to me.

File: keyboards/chocv/keymaps/default/readme.md (hunk 2 above)
Comment (brickbots) [reply keyboards/chocv/keymaps/default/readme.md:9]: Totally willing to change this example as well as the main readme.md , but the Miryoku keymap will be available at build time by virtue of the community layout.  Are you okay if we leave this one as is?

If not, no problem, I'll change it!

File: keyboards/chocv/keymaps/default/config.h (hunk 3 above)
Comment (brickbots) [reply keyboards/chocv/keymaps/default/config.h:19]: Great!  It's gone 👍

File: keyboards/chocv/keymaps/default/readme.md (hunk 2 above)
Comment (precondition) [reply keyboards/chocv/keymaps/default/readme.md:9]: You can include both `make` commands. "Run this make command to compile the very basic test firmware : xxx. Run this make command to compile the 'real' default keymap: yyy"

Comment (brickbots) [reply keyboards/chocv/keymaps/default/readme.md:20]: That's a great idea, I've made the keymap readme more informative and included both make commands.  Thank you @precondition !

File: keyboards/chocv/keymaps/default/config.h (hunk 3 above)
Comment (drashna) [reply keyboards/chocv/keymaps/default/config.h:19]: It's still here.

Comment (brickbots) [reply keyboards/chocv/keymaps/default/config.h:19]: Wow, all I can say is 'Whoops'.  Sorry about the mix up and time sink here.  Thanks both for you time.

**Comment by manna-harbour** [comment]
While this PR is awaiting review, Miryoku QMK can be built for `chocv` directly from the [build workflows](https://github.com/manna-harbour/miryoku_qmk/tree/miryoku/users/manna-harbour_miryoku#workflow-builds) as they can now merge branches at build time.  Use `brickbots/qmk_firmware/chocv` with the `Merge QMK Branches` field / `merge` option to merge the PR branch.

Here's a run for `chocv` with all combinations of Alphas, Nav, and Clipboard alternative layout options: https://github.com/manna-harbour/miryoku_qmk/actions/runs/2176130295.

File: keyboards/chocv/config.h [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+ /* COL2ROW, ROW2COL*/
+ #define DIODE_DIRECTION COL2ROW
+
+
+ /* Debounce reduces chatter (unintended double-presses) - set 0 if debouncing is not needed */
+ #define DEBOUNCE 5
+
+ /* disable these deprecated features by default */
+ #define NO_ACTION_MACRO
+ #define NO_ACTION_FUNCTION
```
Comment (drashna) [inline keyboards/chocv/config.h:46]:
```suggestion
```

File: keyboards/chocv/rules.mk [hunk 5]
```diff
@@ -0,0 +1,14 @@
+# MCU name
+MCU = atmega32u4
+BOOTLOADER = qmk-dfu
```
Comment (drashna) [inline keyboards/chocv/rules.mk:3]:
```suggestion
# Bootloader selection

BOOTLOADER = qmk-dfu
```

File: keyboards/chocv/rules.mk [hunk 6]
```diff
@@ -0,0 +1,14 @@
...
+
+# Build Options
+#   change yes to no to disable
+#
+BOOTMAGIC_ENABLE = no       # Virtual DIP switch configuration
+MOUSEKEY_ENABLE = yes       # Mouse keys
+EXTRAKEY_ENABLE = yes       # Audio control and System control
+CONSOLE_ENABLE = no        # Console for debug
+COMMAND_ENABLE = no        # Commands for debug and configuration
```
Comment (drashna) [inline keyboards/chocv/rules.mk:12]:
```suggestion
BOOTMAGIC_ENABLE = yes      # Enable Bootmagic Lite
MOUSEKEY_ENABLE = yes       # Mouse keys
EXTRAKEY_ENABLE = yes       # Audio control and System control
CONSOLE_ENABLE = no         # Console for debug
COMMAND_ENABLE = no         # Commands for debug and configuration
NKRO_ENABLE = yes           # Enable N-Key Rollover
BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
AUDIO_ENABLE = no           # Audio output
```

**Review by drashna — CHANGES_REQUESTED** [review]

**Comment by brickbots** [comment]
Thanks @drashna for the review.  I'll work to find a more modern/recent example to start from for my next board so it's closer to current standards right off the bat.  Thank you so much!

**Comment by drashna** [comment]
`qmk new-keyboard` is the best way, as it uses the built in template.

**Review by drashna — APPROVED** [review]

File: keyboards/chocv/rules.mk [hunk 7]
```diff
@@ -0,0 +1,20 @@
+# MCU name
+MCU = atmega32u4
+# Bootloader selection
+
```
Comment (fauxpark) [inline keyboards/chocv/rules.mk:4]:
```suggestion

# Bootloader selection
```

**Comment by brickbots** [comment]
Thanks @drashna and @fauxpark for helping to get this PR merged.  I really appreciate the help from both of you.

File: keyboards/chocv/readme.md [hunk 8]
```diff
@@ -0,0 +1,29 @@
...
+
+See the [build environment setup](https://docs.qmk.fm/#/getting_started_build_tools) and the [make instructions](https://docs.qmk.fm/#/getting_started_make_guide) for more information. Brand new to QMK? Start with our [Complete Newbs Guide](https://docs.qmk.fm/#/newbs).
+
+## Bootloader
+
+Enter the bootloader using the hardware reset button on the top middle of the
+near the MCU
+
+
+
```
Comment (noroadsleft) [inline keyboards/chocv/readme.md:29]:
```suggestion
## Bootloader

Enter the bootloader using the hardware reset button on the top middle of the
near the MCU.

See the [build environment setup](https://docs.qmk.fm/#/getting_started_build_tools) and the [make instructions](https://docs.qmk.fm/#/getting_started_make_guide) for more information. Brand new to QMK? Start with our [Complete Newbs Guide](https://docs.qmk.fm/#/newbs).
```

**Review by noroadsleft — CHANGES_REQUESTED** [review]

**Review by fauxpark — APPROVED** [review]

**Review by noroadsleft — APPROVED** [review]

**Comment by noroadsleft** [comment]
Thanks!

**Pull request merged**